examples/eddying_channel.jl

```diff
@@ -0,0 +1,427 @@
+                                                           halo = (3, 3, 3),
+                                                           x = (-Lx/2, Lx/2),
+                                                           y = (0, Ly),
+                                                           z_faces = z_faces)
+
+# We visualize the cell interfaces by plotting the cell height as a function of depth,
+
+plot(underlying_grid.Δzᵃᵃᶜ[1:Nz], underlying_grid.zᵃᵃᶜ[1:Nz],
+     marker = :circle,
+     ylabel = "Depth (m)",
+     xlabel = "Vertical spacing (m)",
+     legend = nothing)
+
+const bump_amplitude = 300meters
+const bump_x_width = 100kilometers
+const bump_y_width = 200kilometers
+
+# We include some non-trivial batymetry in the form of Gaussian hill elongated in the meridional
+# direction. To do that, we construct an `ImmersedBoundaryGrid`,
+
+@inline pow2(x) = x * x
+@inline bump(x, y) = bump_amplitude * exp(-pow2(x) / (2 * pow2(bump_x_width)) - pow2(y - Ly/2) / (2 * pow2(bump_y_width)))
+
+@inline below_bottom(x, y, z) = z < -Lz + bump(x, y)
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBoundary(below_bottom))
+
+# Let's have a look at the bathymetry we've just included.
+
+x, y, z = nodes((Face, Face, Face), grid)
+
+heatmap(x / kilometers, y / kilometers, [bump(x[i], y[j]) for i=1:Nx, j=1:Ny+1]',
+              color = :deep,
+        aspectratio = 1,
+             xlabel = "x [km]",
+             ylabel = "y [km]",
+     colorbar_title = "bathymetry [m]",
+              xlims = (-Lx/2 / kilometers, Lx/2 / kilometers),
+              ylims = (0, Ly / kilometers))
+
+# # Boundary conditions
+#
+# A channel-centered jet and overturning circulation are driven by wind stress and an alternating
+# pattern of surface cooling and surface heating with parameters
+
+Qᵇ = 1e-8            # buoyancy flux magnitude [m² s⁻³]
+y_shutoff = 5/6 * Ly # shutoff location for buoyancy flux [m]
+τ = 1e-4             # surface kinematic wind stress [m² s⁻²]
+μ = 1 / 100days      # bottom drag damping time-scale [s⁻¹]
+
+# The buoyancy flux has a sinusoidal pattern in `y`,
+
+@inline buoyancy_flux(x, y, t, p) = ifelse(y < p.y_shutoff, p.Qᵇ * cos(3π * y / p.Ly), 0)
+
+buoyancy_flux_bc = FluxBoundaryCondition(buoyancy_flux, parameters=(Ly=grid.Ly, y_shutoff=y_shutoff, Qᵇ=Qᵇ))
+
+# At the surface we impose a wind stress with sinusoidal variation in `y`,
+
+@inline u_stress(x, y, t, p) = - p.τ * sin(π * y / p.Ly)
+
+u_stress_bc = FluxBoundaryCondition(u_stress, parameters=(τ=τ, Ly=grid.Ly))
+
+# Linear bottom drag on `u` and `v` provides a sink of momentum
+
+@inline u_drag(x, y, t, u, μ) = - μ * u
+@inline v_drag(x, y, t, v, μ) = - μ * v
+
+u_drag_bc = FluxBoundaryCondition(u_drag, field_dependencies=:u, parameters=μ)
+v_drag_bc = FluxBoundaryCondition(v_drag, field_dependencies=:v, parameters=μ)
+
+# To summarize,
+
+b_bcs = FieldBoundaryConditions(top = buoyancy_flux_bc)
+u_bcs = FieldBoundaryConditions(top = u_stress_bc)
```

Comment:
@glwagner, shouldn't it be

```suggestion
u_bcs = FieldBoundaryConditions(top = u_stress_bc, bottom = u_drag_bc)
```

?